spec/gruf/prometheus/server/collector_spec.rb

```diff
@@ -0,0 +1,209 @@
+  end
+  let(:timed_result) { Gruf::Interceptors::Timer::Result.new(result, 2.0, true) }
+
+  describe '#started_total' do
+    subject { collector.started_total(request: request) }
+
+    it 'pushes the grpc_server_started_total to the server' do
+      expect(collector).to receive(:push).with(
+        grpc_server_started_total: 1,
+        custom_labels: {
+          grpc_method: 'GetThing',
+          grpc_service: 'gruf.demo.ThingService',
+          grpc_type: 'UNARY'
+        }
+      )
+      subject
+    end
+  end
+
+  describe '#handled_total' do
+    subject { collector.handled_total(request: request, result: timed_result) }
+
+    it 'pushes the grpc_server_handled_total to the server' do
+      expect(collector).to receive(:push).with(
+        grpc_server_handled_total: 1,
+        custom_labels: {
+          grpc_method: 'GetThing',
+          grpc_service: 'gruf.demo.ThingService',
+          grpc_type: 'UNARY',
+          grpc_code: 'OK'
+        }
+      )
+      subject
+    end
+
+    context 'when the request is unsuccessful' do
+      let(:result) { GRPC::Internal.new('fail') }
+      let(:timed_result) { Gruf::Interceptors::Timer::Result.new(result, 2.0, false) }
+
+      it 'pushes the grpc_server_handled_total to the server with an unsuccessful code' do
+        expect(collector).to receive(:push).with(
+          grpc_server_handled_total: 1,
+          custom_labels: {
+            grpc_method: 'GetThing',
+            grpc_service: 'gruf.demo.ThingService',
+            grpc_type: 'UNARY',
+            grpc_code: 'Internal'
+          }
+        )
+        subject
+      end
+    end
+  end
+
+  describe '#handled_latency_seconds' do
+    subject { collector.handled_latency_seconds(request: request, result: timed_result) }
+
+    it 'pushes the grpc_server_handled_latency_seconds to the server' do
+      expect(collector).to receive(:push).with(
+        grpc_server_handled_latency_seconds: 2.0,
+        custom_labels: {
+          grpc_method: 'GetThing',
+          grpc_service: 'gruf.demo.ThingService',
+          grpc_type: 'UNARY',
+          grpc_code: 'OK'
+        }
+      )
+      subject
+    end
+
+    context 'when the request is unsuccessful' do
+      let(:result) { GRPC::Internal.new('fail') }
+      let(:timed_result) { Gruf::Interceptors::Timer::Result.new(result, 2.0, false) }
+
+      it 'pushes the grpc_server_handled_latency_seconds to the server with an unsuccessful code' do
+        expect(collector).to receive(:push).with(
+          grpc_server_handled_latency_seconds: 2.0,
+          custom_labels: {
+            grpc_method: 'GetThing',
+            grpc_service: 'gruf.demo.ThingService',
+            grpc_type: 'UNARY',
+            grpc_code: 'Internal'
+          }
+        )
+        subject
+      end
+    end
+  end
+
+  describe '.custom_labels' do
+    subject { collector.started_total(request: request) }
+
+    context 'when the service is a top-level service' do
```

Comment:
❓ What is an alternative to being a `top-level service`?

Reply:
The other examples: `gruf.demo.ThingService`, as opposed to `ThingService`